src/components/TileEditor.tsx

```diff
@@ -1,14 +1,78 @@
 import React from 'react';
-import { TileInfo } from '../firebase';
+import { firestore, TileInfo } from '../firebase';
+import { useForm } from "react-hook-form";
 
 export function TileEditor(props: {tileInfo: TileInfo, tileName: string}) {
+  const { register, setValue, handleSubmit, errors } = useForm<TileInfo>();
+  const onSubmit = handleSubmit( async ({ bg_color, bg_image, hover_text, description_text, link }) => {
+    const tile = firestore.collection("tiles").doc(props.tileName);
+    try{const result = await tile.update({ 
+        bg_color: bg_color,
+        bg_image: bg_image,
+        hover_text: hover_text,
+        description_text: description_text,
+        link: link
+      });      
+    } catch {
+      console.log("Update failed")
+    }
+  });
+
   return (
-    <div>
-      <h2>Tile: {props.tileName}</h2>
-      <h3>Hover Text: {props.tileInfo.hover_text}</h3>
-      <h3>Description: {props.tileInfo.description_text}</h3>
-    </div>
+    <form onSubmit={onSubmit}>
+      <h2>{props.tileName}</h2>
+      <label>Color
+        <input 
+          name="bg_color"
+          type="text"
+          key={`bg_color:${props.tileInfo.bg_color}`}
+          defaultValue={props.tileInfo.bg_color}
+          ref={register}
+        />
+      </label>
+      <br />
+      <label>Image
+        <input
+          name="bg_image"
+          type="text"
+          key={`bg_image:${props.tileInfo.bg_image}`}
+          defaultValue={props.tileInfo.bg_image}
+          ref={register}
+        />
+      </label>
+      <br />
+      <label>Hover Text
+        <input
+          name="hover_text"
+          type="text"
+          key={`hover_text:${props.tileInfo.hover_text}`}
+          defaultValue={props.tileInfo.hover_text}
+          ref={register}
+          />
+      </label>
+      <br />
+      <label>Description Text
+        <input
```

Comment:
This probably wants to be a <textarea> so Shannon has more space to write.

Reply:
Yeah I tried that initially and weirdly it didn't seem to make a difference? I can poke at it more and see if it's a stlyling thing or something, but I figured this works for now.

Reply:
Nevermind, I figured out how I was derping it. It's a textarea now.